### PR TITLE
Improve query pass to wazuhdb

### PR DESCRIPTION
**Pull request by GabrielEValenzuela**
# Objective

This PR aims to add a validation step for the incoming query sent from analysisd to the wazuhdb

### CODE REVIEW

File: src/analysisd/decoders/dbsync.c [hunk 1]
```diff
@@ -173,6 +187,13 @@ static void dispatch_state(dbsync_context_t * ctx) {
     os_malloc(OS_MAXSTR, query);
     os_malloc(OS_MAXSTR, response);
 
+    ComponentType component_type = validate_component(ctx->component);
```
Comment (pereyra-m) [inline src/analysisd/decoders/dbsync.c:190]: We can remove this and centralize the component check at `DispatchDBSync()`

File: src/analysisd/decoders/dbsync.c [hunk 2]
```diff
@@ -219,6 +240,13 @@ static void dispatch_clear(dbsync_context_t * ctx) {
...
```
Comment (pereyra-m) [inline src/analysisd/decoders/dbsync.c:243]: We can remove this and centralize the component check at `DispatchDBSync()`

File: src/analysisd/decoders/dbsync.c [hunk 3]
```diff
@@ -123,6 +130,13 @@ static void dispatch_check(dbsync_context_t * ctx, const char * command) {
...
```
Comment (pereyra-m) [inline src/analysisd/decoders/dbsync.c:133]: We can remove this and centralize the component check at `DispatchDBSync()`

File: src/analysisd/decoders/dbsync.c [hunk 4]
```diff
@@ -92,6 +92,13 @@ static void dispatch_answer(dbsync_context_t * ctx, const char * result) {
     char * query;
     os_malloc(OS_MAXSTR, query);
 
+    ComponentType component_type = validate_component(ctx->component);
```
Comment (pereyra-m) [inline src/analysisd/decoders/dbsync.c:95]: We can remove this and centralize the component check at `DispatchDBSync()`

File: src/wazuh_db/wdb_shared.c [hunk 5]
```diff
@@ -25,6 +25,29 @@ const char* WDBC_RESULT[] = {
     [WDBC_UNKNOWN] = "unk"
 };
 
+/// Array of component strings.
+const char* wdbc_valid_components[] = {
```
Comment (pereyra-m) [inline src/wazuh_db/wdb_shared.c:29]: Please, use the same trick as `WDBC_RESULT`, where each value is already tied to the enum value

File: src/shared/wazuhdb_op.c [hunk 6]
```diff
@@ -201,6 +201,20 @@ int wdbc_parse_result(char *result, char **payload) {
     return retval;
 }
 
+/**
+ * @brief Validate the component of the query.
+ * @param[in] component Component to be validated.
+ * @return Enum ComponentType. COMPONENT_INVALID if the component is not valid.
+ */
+ComponentType validate_component(const char *component) {
```
Comment (pereyra-m) [inline src/shared/wazuhdb_op.c:209]: All the functions in this file have the prefix `wdbc_`

File: src/headers/wazuhdb_op.h [hunk 7]
```diff
@@ -19,22 +19,52 @@
...
+    CLEANDATABASE,
+    SCAN_INFO_UPDATE,
+    CONTROL,
+    LOAD,
+    COMP_DELETE,
+    SAVE,
+    SAVE2,
+    COMP_INTEGRITY_CLEAR,
+    COMPONENT_INVALID // Fallback for invalid values
+} ComponentType;
```
Comment (pereyra-m) [inline src/headers/wazuhdb_op.h:46]: We could use `component_type` to be consistent with the naming of the other enum

File: src/unit_tests/shared/test_wazuhdb_op.c
Comment (pereyra-m) [inline src/unit_tests/shared/test_wazuhdb_op.c:1]: It's better not to change the style of the methods not modified in the PR.

File: src/wazuh_db/wdb_shared.c [hunk 8]
```diff
@@ -25,6 +25,29 @@ const char* WDBC_RESULT[] = {
     [WDBC_UNKNOWN] = "unk"
 };
 
+/// Array of component strings.
+const char* wdbc_valid_components[] = { 
+    "syscollector_processes",
```
Comment (pereyra-m) [inline src/wazuh_db/wdb_shared.c:30]: Considering that DBSync/RSync can be used by `syscheck`, this the list of components that trigger `wdb_parse_syscheck()` and end up managing `state`, `integrity_check_` and `integrity_clear` commands

<details><summary>Details</summary>
<p>





</p>
</details> 

The same for `syscollector`

<details><summary>Details</summary>
<p>




</p>
</details>

**Review by pereyra-m — CHANGES_REQUESTED** [review]
Gj!

File: src/analysisd/decoders/dbsync.c (hunk 1 above)
Comment (GabrielEValenzuela) [reply src/analysisd/decoders/dbsync.c]: Moved!

File: src/analysisd/decoders/dbsync.c (hunk 3 above)
Comment (GabrielEValenzuela) [reply src/analysisd/decoders/dbsync.c]: Moved!

File: src/analysisd/decoders/dbsync.c (hunk 4 above)
Comment (GabrielEValenzuela) [reply src/analysisd/decoders/dbsync.c]: Moved!

File: src/wazuh_db/wdb_shared.c (hunk 5 above)
Comment (GabrielEValenzuela) [reply src/wazuh_db/wdb_shared.c]: Done!

File: src/shared/wazuhdb_op.c (hunk 6 above)
Comment (GabrielEValenzuela) [reply src/shared/wazuhdb_op.c]: Fixed!

File: src/headers/wazuhdb_op.h (hunk 7 above)
Comment (GabrielEValenzuela) [reply src/headers/wazuhdb_op.h]: Done!

File: src/unit_tests/shared/test_wazuhdb_op.c
Comment (GabrielEValenzuela) [reply src/unit_tests/shared/test_wazuhdb_op.c:1]: Restored, the clang-format of VsCode

File: src/wazuh_db/wdb_shared.c (hunk 8 above)
Comment (GabrielEValenzuela) [reply src/wazuh_db/wdb_shared.c]: I made an upgrade of the list, please tell me if it's ok. Tomorrow we can sync to verify if I understand your images correctly

Comment (pereyra-m) [reply src/wazuh_db/wdb_shared.c]: Ok , thank you.

**Review by pereyra-m — APPROVED** [review]
Great job!

I've reviewed the code.
To verify everything is working, I've installed the development branch and connected two agents v4.9.2 (Ubuntu 22 and Windows 10).

After the `syscollector` and `syscheck` scan , no error or warning messages were found



Also, some invalid messages were sent and now are rejected as expected.

**Review by sebasfalcone — APPROVED** [review]
## Review
- Code 🟢 
  - Comprehensive and easy to maintain 
  
- Functional 🟢 
  - Tested [here](https://github.com/wazuh/internal-devel-requests/issues/1749#issuecomment-2462356542) with a custom agent which dispatches malicious requests
  - New integration tests added to validate changes

**Pull request merged**